public/app/features/alerting/unified/mocks/server/db.ts

```diff
@@ -10,40 +10,37 @@ import {
   PromRuleGroupDTO,
   PromRuleType,
   RulerAlertingRuleDTO,
+  RulerCloudRuleDTO,
   RulerRecordingRuleDTO,
   RulerRuleGroupDTO,
 } from 'app/types/unified-alerting-dto';
 
 import { MockDataSourceSrv } from '../../mocks';
 import { DataSourceType } from '../../utils/datasource';
+import { namespaces } from '../mimirRulerApi';
 
-const prometheusRuleFactory = Factory.define<PromAlertingRuleDTO>(({ sequence }) => ({
-  name: `test-rule-${sequence}`,
+import { MIMIR_DATASOURCE_UID, PROMETHEUS_DATASOURCE_UID } from './constants';
+
+class PromRuleFactory extends Factory<PromAlertingRuleDTO> {
+  fromRuler(rulerRule: RulerAlertingRuleDTO) {
+    return this.params({
+      name: rulerRule.alert,
+      query: rulerRule.expr,
+      type: PromRuleType.Alerting,
+      labels: rulerRule.labels,
+      annotations: rulerRule.annotations,
+    });
+  }
+}
```

Comment:
How do you feel about this pattern and using fishery for building mocks?

Reply:
I love it! use a `.fromX().build() `in case you have the object or the` .build()` directly...nice!

Reply:
Overall I think I like it, I do think we have a few slightly different approaches overall - but if the end result is we have a mock API that behaves in the right way, then it doesn't matter _too_ much how it's done